packages/block-library/src/query-loop/edit.js

```diff
@@ -17,12 +17,12 @@ import { useQueryContext } from '../query';
 const TEMPLATE = [ [ 'core/post-title' ], [ 'core/post-content' ] ];
 export default function QueryLoopEdit( {
 	clientId,
-	context: {
-		query: { perPage, offset, categoryIds },
-		queryContext,
-	},
+	context: { query: _query, queryContext },
```

Comment:
Won't query default to an empty object?

Reply:
It seems to be undefined when loading in the post editor ( Appearance -> Templates -> some template with a query loop ).  So being de-structured at that previous level was crashing with a `cannot find perPage of undefined` type error.

Reply:
Can we add a default value to the attribute? It would be simpler.

Reply:
Im familiar with how to do this for attributes in the block.json, but how do we do it for items in the `usesContext` array? 🤔

Reply:
In the parameter destructuring.

Reply:
Ohhh... gotcha.  So more like:
```
( {  context: { query = {}, queryContext } } )
const { perPage, offset, catergoryIds } = query;
```
^^ this?

Or are you mentioning there is a way to apply a default and continue to destructure from it in that perPage, offset, categoryIds would still be defined in that destructuring as well?  LOL, it would be nice but im not sure how that would look but a not-working example being:
```
( {  context: { query = {}: { perPage, offset, categoryIds }, queryContext } } )
```

Reply:
Updated for the first, but it would be neat if there was a way to do the second. 😆

Reply:
`( {
	context: { query: { perPage, offset, categoryIds } = {}, queryContext },
} )`

Reply:
Oo yes, that does it.  Thank you!